### ISSUE TITLE
attach

### ISSUE THREAD

**Issue by rosscdh** [SkullTech/drymail#2]
Hey there nice lib thanks

found a small bug in the example

```
client = SMTPMailer(host='smtp.gmail.com', user='name@example.com', password='a', tls=True)
message = Message(subject='Congrats :)', sender=('Showcase', 'name@example.com'),
                  receivers=['name@example2.com'], text='When is the party? ;)')

with open('/tmp/textfile.txt', 'rb') as pdf_file:
    message.attach(filename='/tmp/textfile.txt', data=pdf_file.read(), mimetype='application/pdf')
```

throws an error about attaching to a None object where self.message is  None

need to call 

```
message.prepare()
```

Before the attach..

Recommend moving this call to the __init__ so that it removes this step from the user.

**Comment by skulltech** [comment]
Thanks for pointing out the bug! @rosscdh
 
You're right, `message.prepare` is kind of weak point of the architecture. Calling it inside `__init__` would make it a lot more straightforward, but I wanted to let the user set various parameters themselves, even after instantiating the object.  That was the motivation behind making a different function `prepare` and _not_ call it in `__init__` itself.  But then again, this adds an extra step for the user, which is bad. That's why I made SMTPMailer check if the message is _prepared_ or not before sending it. That way I was able to find a middle ground.

I'll push a commit soon to tackle this issue, I think I'll just initialize `self.message` to `MIMEMultipart('mixed')` in `__init__` for now. But I need to find a more elegant method to deal with this, probably for a future release.
